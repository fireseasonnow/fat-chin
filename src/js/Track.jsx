import React from 'react';
import styled from 'styled-components';

const Track = props => (
    <Item>
        <Button onClick={() => props.showLyricsHandler(props.name)} type='button'>
            {props.number}. {props.name}
        </Button>
    </Item>
);

export default Track;

const Item = styled.li`
    position: relative;
    cursor: pointer;
    list-style: none;

    &:hover::after {
        content: 'lyrics';
        font-size: 10px;
        position: absolute;
        right: 0;
        bottom: 0;
    }
`;

const Button = styled.button`
    background: none;
    color: inherit;
    font-size: inherit;
    font-family: inherit;
    border: 0;
    padding: 0;
    cursor: inherit;
    outline: none;
    text-align: left;
`;
