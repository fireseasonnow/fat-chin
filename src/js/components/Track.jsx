// @flow
import React from 'react';

type Props = {
    number: number,
    name: string,
    showLyricsHandler: (e: Object, name: string) => void
};

const track = (props: Props) => (
    <li className="album__track"><button className="album__track-btn qa-album__track-btn" type="button" onClick={(e) => props.showLyricsHandler(e, props.name)}>{props.number}. {props.name}</button></li>
);

export default track;