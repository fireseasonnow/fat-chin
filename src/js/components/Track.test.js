import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';

import Track from './Track';

configure({ adapter: new Adapter() });

describe('<Track />', () => {
    let wrapper = {};

    beforeEach(() => {
        wrapper = shallow(<Track />);
    });

    test('should render without crashing', () => {
        expect(wrapper).toBeTruthy();
    });

    test('should render a track with name and number', () => {
        wrapper.setProps({
            number: 1,
            name: 'Hello',
        });

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    test('should fire showLyricsHandler when clicked on a button', () => {
        const showLyricsHandler = jest.fn();

        wrapper.setProps({
            name: 'Hello',
            showLyricsHandler,
        });

        const event = { target: { value: 'test' } };
        wrapper.find('Track__Button').simulate('click', event.target.value, 'Hello');

        expect(showLyricsHandler).toHaveBeenCalledWith(event.target.value, 'Hello');
    });
});
