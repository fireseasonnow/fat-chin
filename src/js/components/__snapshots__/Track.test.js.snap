// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Track /> should render a track with name and number 1`] = `
<li
  className="album__track"
>
  <button
    className="album__track-btn qa-album__track-btn"
    onClick={[Function]}
    type="button"
  >
    1
    . 
    Hello
  </button>
</li>
`;
